src/utils/user-role-utils.tsx

```diff
@@ -0,0 +1,31 @@
+import { useAtomValue } from "jotai";
+import { KeycloakTokenParsed } from "keycloak-js";
+import { authAtom } from "../atoms/auth";
+import { useLocation } from "react-router";
+
+/**
+ * User Role Utils class
+ */
+export default class UserRoleUtils {
+  /**
+   * Check if the logged-in user has an admin role
+   *
+   * @param accessToken keycloak access token
+   * @returns boolean
+   */
+  public static isAdmin = (accessToken?: KeycloakTokenParsed) => {
+    return !!accessToken?.realm_access && accessToken.realm_access?.roles.includes("admin");
```

Comment:
I'm curious about this double exclamation mark. Does it negate "not" / ! accessToken, therefore is it the same as straight up using accessToken?.realm_access without any negations?

Reply:
Good question, I just copied this bit from the old time-bank-ui. Maybe this could be simplified.

Reply:
I read that the double negation checks whether a value exists or not. It is the same as doing `variable != null ? true : false`

Reply:
Not sure if I'm a fan of that. It's basically doing a impromptu cast to bool via double negation. It's... terrifying. And quite ambiguous looking.

Reply:
I got rid of the double negation. It should be easier to read now.
`  public static isAdmin = () => {
    const accessToken = useAtomValue(authAtom)?.token;
    if (!accessToken) return false;

    return accessToken.realm_access ? accessToken.realm_access.roles.includes("admin") : false;
  };`